### PR TITLE
[17.0][MIG] event_stage_cancelled: Migration to 17.0

### PR DESCRIPTION
**Pull request by victoralmau**
Migration to 17.0

Please @chienandalu and @pilarvargas-tecnativa can you review it?

@Tecnativa TT51925

### CODE REVIEW

File: event_stage_cancelled/data/mail_template_data.xml [hunk 1]
```diff
@@ -179,6 +179,6 @@
 </td></tr>
 </table>
         </field>
-        <field name="lang">{{ object.partner_id.lang }}</field>
+        <field name="lang">{{ object.event_id.lang or object.partner_id.lang }}</field>
```
Comment (chienandalu) [inline event_stage_cancelled/data/mail_template_data.xml:182]: Why to choose event lang over partner's?

Comment (victoralmau) [reply event_stage_cancelled/data/mail_template_data.xml:182]: It is to be consistent with what core does in the event email templates.

**Review by chienandalu — APPROVED** [review]
Test and code review :+1:

File: event_stage_cancelled/tests/test_event_stage_cancelled.py [hunk 2]
```diff
@@ -65,7 +65,7 @@ def test_event_cancellation(self):
         self.assertFalse(bool(self.event_mail.mail_registration_ids))
         self.assertFalse(self.event_mail.scheduled_date)
         self.assertEqual(self.event_mail.mail_state, "running")
-        # Inject bypass_reason for test compatibility with event_registration_cancel_reason
+        # Inject bypass_reason for test compatibility with event_registration_cancel_reason  # noqa: E501
```
Comment (pilarvargas-tecnativa) [inline event_stage_cancelled/tests/test_event_stage_cancelled.py]: In this case I think it is not necessary to ignore the linter as it is a simple comment and does not influence the logic. Split the comment and that's it

**Review by pilarvargas-tecnativa — APPROVED** [review]

**Pull request merged**